### ISSUE TITLE
The randomness of the training process

### ISSUE THREAD

**Issue by MelanTech** [LitingLin/LoRAT#13]
Hello! I encountered a rather tricky problem while conducting relevant training work and would like to consult with you for a solution.

During the training process, I always ensured that the training parameters were completely consistent. However, the loss values generated from training showed inconsistency, which prevented me from reproducing consistent training results under the same settings.

For this purpose, I found the enable_deterministic_computation() function in the `trackit/core/boot/workarounds/reproducibility.py` file. I called this function during the initialization phase, but unfortunately, this did not achieve the expected effect and did not eliminate the differences in training results.

After analysis, I found that the root of randomness lies in the dataloader in the data loading process. During each training process, the data loaded is not consistent, which introduces randomness factors and interferes with the reproducibility of the training results.

Given the current dilemma, I would like to ask what effective measures should be taken to eliminate the randomness caused by dataloaders, in order to ensure consistent training results under the same training settings? Looking forward to your professional guidance and valuable suggestions. Thank you very much!

**Comment by MelanTech** [comment]
I have found a solution, just configure seed for the sampler in `run.yaml`:
```yaml
  sampler:
    type: "random"

    samples_per_epoch: 131072
    seed: 42   # Set seed here
    ...
```

**Comment by LitingLin** [comment]
I managed to improve the training reproducibility in recent commits. Sorry for the inconvenience.
